wnfs-wasm/src/fs/private/file.rs

```diff
@@ -1,42 +1,87 @@
 //! The bindgen API for PrivateFile.
 
+use crate::{
+    fs::{
+        utils::{self, error},
+        BlockStore, ForeignBlockStore, JsResult, Namefilter, PrivateForest, Rng,
+    },
+    value,
+};
 use chrono::{DateTime, Utc};
-use js_sys::Date;
-use wasm_bindgen::prelude::wasm_bindgen;
+use js_sys::{Date, Promise, Uint8Array};
+use std::rc::Rc;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+use wasm_bindgen_futures::future_to_promise;
 use wnfs::{Id, PrivateFile as WnfsPrivateFile};
 
-use crate::fs::{JsResult, Namefilter, Rng};
-
 //--------------------------------------------------------------------------------------------------
 // Type Definitions
 //--------------------------------------------------------------------------------------------------
 
 /// A file in a WNFS public file system.
 #[wasm_bindgen]
-pub struct PrivateFile(WnfsPrivateFile);
+pub struct PrivateFile(pub(crate) Rc<WnfsPrivateFile>);
 
 //--------------------------------------------------------------------------------------------------
 // Implementations
 //--------------------------------------------------------------------------------------------------
 
 #[wasm_bindgen]
 impl PrivateFile {
-    /// Creates a new private file.
-    #[wasm_bindgen(constructor)]
-    pub fn new(
+    /// Creates an empty private file.
+    pub fn empty(parent_bare_name: Namefilter, time: &Date, mut rng: Rng) -> JsResult<PrivateFile> {
```

Comment:
minor: this still feels like new to me, even though it's essentially empty to start, b/c in Rust-land new can also mean an empty hashmap etc, but just thinking aloud.